c++/include/orc/Type.hh

```diff
@@ -58,6 +58,11 @@ namespace orc {
     virtual uint64_t getMaximumLength() const = 0;
     virtual uint64_t getPrecision() const = 0;
     virtual uint64_t getScale() const = 0;
+    virtual Type& setAttribute(const std::string& key,
```

Comment:
What's the benefit to make return type as Type& for setAttribute() and removeAttribute() ? Is it better to make it void?

Reply:
This allows method chaining. I've followed the Java implementation:
https://github.com/apache/orc/blob/f381f2dd78e6b305b8a4e0b07124034de72ef1a1/java/core/src/java/org/apache/orc/TypeDescription.java#L242-L252
But I can switch to void if you prefer that.